src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -51,11 +55,35 @@ object RuleUtils {
       }
     }
 
+    def isHybridScanCandidate(entry: IndexLogEntry, files: Set[FileInfo]): Boolean = {
+      // compare all the metadata of source files
+      assert(entry.relations.length == 1)
+      // TODO threshold for the number of CommonFiles
+      // TODO plan signature
+      files.exists(entry.allSourceFileInfos.contains)
+    }
+
     // TODO: the following check only considers indexes in ACTIVE state for usage. Update
```

Comment:
I remember there is already an issue for this. If so, can we just link to the issue containing the description. @apoorvedave1 to confirm if there is an issue.

Reply:
It seems #65; I'll add the link here.